encode_test.go

```diff
@@ -101,6 +85,29 @@ func TestParseTs(t *testing.T) {
 	}
 }
 
+var timeErrorTests = []string{
+	"2001",
+	"2001-2-03",
+	"2001-02-3",
+	"2001-02-03 ",
+	"2001-02-03 04",
+	"2001-02-03 04:",
+	"2001-02-03 04:05",
+	"2001-02-03 04:05:",
+	"2001-02-03 04:05:6",
+	"2001-02-03 04:05:06.123 B",
```

Comment:
These are values which would cause a panic before. I originally made this change in the [cockroach fork](https://github.com/cockroachdb/pq/commit/17d569dbc13f29cbe32b50e64d2046648cd9684e) because it was being called from a place where panics were unacceptable and the overhead of wrapping it in a deferred recover was too much.

I ran all the benchmarks and nothing changed except this one:

    name                 old time/op    new time/op    delta
    DecodeTimestamptz-8     728ns ± 2%     760ns ± 1%  +4.36%        (p=0.000 n=13+13)
    
    name                 old alloc/op   new alloc/op   delta
    DecodeTimestamptz-8     64.0B ± 0%     64.0B ± 0%    ~     (all samples are equal)
    
    name                 old allocs/op  new allocs/op  delta
    DecodeTimestamptz-8      2.00 ± 0%      2.00 ± 0%    ~     (all samples are equal)